src/helpers/messages.js

```diff
@@ -57,23 +57,21 @@ function getUnansweredItemsWarning(scope, runner, sync) {
     var itemsCountMessage = '';
 
     if (scope === 'section' || scope === 'testSection') {
-        if (unansweredCount === 0) {
-            itemsCountMessage = __('You answered all %s question(s) in this section', stats.questions.toString());
-        } else {
-            itemsCountMessage = __(
-                'You answered only %s of the %s question(s) in this section',
-                stats.answered.toString(),
-                stats.questions.toString()
-            );
-        }
+        itemsCountMessage = `<b>${__('You are about to leave this section.')}</b><br><br>`
+        itemsCountMessage += __(
+            'You answered %s of %s question(s) for this section of the test',
+            stats.answered.toString(),
+            stats.questions.toString()
+        );
         if (flaggedCount) {
             itemsCountMessage += `, ${__('and flagged %s of them', flaggedCount.toString())}`;
         }
     } else if (scope === 'test' || scope === 'testWithoutInaccessibleItems') {
-        if (unansweredCount === 0) {
-            itemsCountMessage = __('You answered all %s question(s) in this test', stats.questions.toString());
-        } else {
-            itemsCountMessage = __('You have %s unanswered question(s)', unansweredCount.toString());
+        itemsCountMessage = `<b>${__('You are about to submit the test.')}</b><br><br>`
+        if (unansweredCount > 1) {
+            itemsCountMessage += __('There are %s unanswered questions', unansweredCount.toString());
+        } else if (unansweredCount === 1) {
+            itemsCountMessage += __('There is %s unanswered question', unansweredCount.toString());
         }
         if (flaggedCount) {
```

Comment:
```suggestion
        if (unansweredCount && flaggedCount) {
```

Reply:
maybe not show flagged if we don't have unanswered? In this case, we will not have sentence started with `and you flagged `

Reply:
I think flagged behavior is different from if user answer or not